api.planx.uk/modules/flows/validate/service/planningConstraints.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  ComponentType,
+  type FlowGraph,
+} from "@opensystemslab/planx-core/types";
+import { numberOfComponentType } from "../helpers.js";
+import type { FlowValidationResponse } from "./index.js";
+
+const validatePlanningConstraints = (
+  flowGraph: FlowGraph,
+): FlowValidationResponse => {
+  const numberOfPlanningConstraintNodes = numberOfComponentType(
+    flowGraph,
+    ComponentType.PlanningConstraints,
+  );
+  if (numberOfPlanningConstraintNodes > 1) {
+    return {
+      title: "Planning Constraints",
+      status: "Fail",
+      message:
+        "When using Planning Constraints, your flow must have exactly ONE Planning Constraints component",
+    };
+  } else if (numberOfPlanningConstraintNodes === 1) {
+    // In future, add extra `hasPlanningData` validation step here to ensure integration is available for this team
```

Comment:
This is a nice-to-have, but will require team settings context as well as the flattened flow data structure to be available here. The editor preview panel already shows a warning if the integration isn't available for your team, so definitely not urgent!